pkg/resource/user/sdk.go

```diff
@@ -0,0 +1,520 @@
+			}
+			if f5iter.Value != nil {
+				f5elem.Value = f5iter.Value
+			}
+			f5 = append(f5, f5elem)
+		}
+		ko.Spec.Tags = f5
+	} else {
+		ko.Spec.Tags = nil
+	}
+	if resp.User.UserId != nil {
+		ko.Status.UserID = resp.User.UserId
+	} else {
+		ko.Status.UserID = nil
+	}
+	if resp.User.UserName != nil {
+		ko.Spec.Name = resp.User.UserName
+	} else {
+		ko.Spec.Name = nil
+	}
+
+	rm.setStatusDefaults(ko)
+	if err := rm.syncPolicies(ctx, &resource{ko}); err != nil {
+		return nil, err
+	}
+
+	return &resource{ko}, nil
+}
+
+// newCreateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Create API call for the resource
+func (rm *resourceManager) newCreateRequestPayload(
+	ctx context.Context,
+	r *resource,
+) (*svcsdk.CreateUserInput, error) {
+	res := &svcsdk.CreateUserInput{}
+
+	if r.ko.Spec.Path != nil {
+		res.SetPath(*r.ko.Spec.Path)
+	}
+	if r.ko.Spec.PermissionsBoundary != nil {
+		res.SetPermissionsBoundary(*r.ko.Spec.PermissionsBoundary)
+	}
+	if r.ko.Spec.Tags != nil {
+		f2 := []*svcsdk.Tag{}
+		for _, f2iter := range r.ko.Spec.Tags {
+			f2elem := &svcsdk.Tag{}
+			if f2iter.Key != nil {
+				f2elem.SetKey(*f2iter.Key)
+			}
+			if f2iter.Value != nil {
+				f2elem.SetValue(*f2iter.Value)
+			}
+			f2 = append(f2, f2elem)
+		}
+		res.SetTags(f2)
+	}
+	if r.ko.Spec.Name != nil {
+		res.SetUserName(*r.ko.Spec.Name)
+	}
+
+	return res, nil
+}
+
+// sdkUpdate patches the supplied resource in the backend AWS service API and
+// returns a new resource with updated fields.
+func (rm *resourceManager) sdkUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.UpdateUserOutput
+	_ = resp
+	resp, err = rm.sdkapi.UpdateUserWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateUser", err)
+	if err != nil {
+		return nil, err
+	}
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+	if delta.DifferentAt("Spec.Policies") {
+		if err := rm.syncPolicies(ctx, &resource{ko}); err != nil {
+			return nil, err
+		}
+	}
+	if delta.DifferentAt("Spec.Tags") {
+		if err := rm.syncTags(ctx, &resource{ko}); err != nil {
+			return nil, err
+		}
+	}
+
+	return &resource{ko}, nil
+}
+
+// newUpdateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+func (rm *resourceManager) newUpdateRequestPayload(
+	ctx context.Context,
+	r *resource,
+) (*svcsdk.UpdateUserInput, error) {
+	res := &svcsdk.UpdateUserInput{}
+
+	if r.ko.Spec.Name != nil {
+		res.SetUserName(*r.ko.Spec.Name)
```

Comment:
I think you want this to be `res.setNewUserName(*r.ko.Spec.Name)`
and you also want to `res.setUserName()` to the previous name of the user.

https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#UpdateUserInput
See https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#UpdateUserInput.SetNewUserName